In [282]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import datetime as dt
import scipy.stats as stats

In [288]:
loans= pd.read_csv(r"C:\Users\hp\Downloads\Python Case Study\4. Basic Statistics - Hypothesis Testing\LoansData.csv")
loans.head()

Amount.Requested  Amount.Funded.By.Investors Interest.Rate Loan.Length  \
0           20000.0                     20000.0         8.90%   36 months   
1           19200.0                     19200.0        12.12%   36 months   
2           35000.0                     35000.0        21.98%   60 months   
3           10000.0                      9975.0         9.99%   36 months   
4           12000.0                     12000.0        11.71%   36 months   

         Loan.Purpose Debt.To.Income.Ratio State Home.Ownership  \
0  debt_consolidation               14.90%    SC       MORTGAGE   
1  debt_consolidation               28.36%    TX       MORTGAGE   
2  debt_consolidation               23.81%    CA       MORTGAGE   
3  debt_consolidation               14.30%    KS       MORTGAGE   
4         credit_card               18.78%    NJ           RENT   

   Monthly.Income FICO.Range  Open.CREDIT.Lines  Revolving.CREDIT.Balance  \
0         6541.67    735-739               14.0                   14272.0   
1         4583.33    715-719               12.0                   11140.0   
2        11500.00    690-694               14.0                   21977.0   
3         3833.33    695-699               10.0                    9346.0   
4         3195.00    695-699               11.0                   14469.0   

   Inquiries.in.the.Last.6.Months Employment.Length  
0                             2.0          < 1 year  
1                             1.0           2 years  
2                             1.0           2 years  
3                             0.0           5 years  
4                             0.0           9 years

In [292]:
loans.dtypes

Amount.Requested                  float64
Amount.Funded.By.Investors        float64
Interest.Rate                      object
Loan.Length                        object
Loan.Purpose                       object
Debt.To.Income.Ratio               object
State                              object
Home.Ownership                     object
Monthly.Income                    float64
FICO.Range                         object
Open.CREDIT.Lines                 float64
Revolving.CREDIT.Balance          float64
Inquiries.in.the.Last.6.Months    float64
Employment.Length                  object
dtype: object

In [294]:
loans.isna().sum()

Amount.Requested                   1
Amount.Funded.By.Investors         1
Interest.Rate                      2
Loan.Length                        0
Loan.Purpose                       0
Debt.To.Income.Ratio               1
State                              0
Home.Ownership                     1
Monthly.Income                     1
FICO.Range                         2
Open.CREDIT.Lines                  3
Revolving.CREDIT.Balance           3
Inquiries.in.the.Last.6.Months     3
Employment.Length                 77
dtype: int64

In [296]:
loans.columns= loans.columns.str.replace('.','_')

In [298]:
loans['Interest_Rate'] = loans.Interest_Rate.str.replace('%','').astype('float64')

In [300]:
loans.dtypes

Amount_Requested                  float64
Amount_Funded_By_Investors        float64
Interest_Rate                     float64
Loan_Length                        object
Loan_Purpose                       object
Debt_To_Income_Ratio               object
State                              object
Home_Ownership                     object
Monthly_Income                    float64
FICO_Range                         object
Open_CREDIT_Lines                 float64
Revolving_CREDIT_Balance          float64
Inquiries_in_the_Last_6_Months    float64
Employment_Length                  object
dtype: object

In [302]:
loans.duplicated().sum()

0

In [304]:
### Handling the null values of continuous and categorical variables:

def missing_value(x):
    if (x.dtype in ['float64','int64']):
        return x.fillna(x.mean())
        
    else:
        return x.fillna(x.mode().iloc[0])

loans.iloc[:, 0:len(loans.columns)] = loans.iloc[:, 0:len(loans.columns)].apply(missing_value)

In [308]:
loans.isna().sum()

Amount_Requested                  0
Amount_Funded_By_Investors        0
Interest_Rate                     0
Loan_Length                       0
Loan_Purpose                      0
Debt_To_Income_Ratio              0
State                             0
Home_Ownership                    0
Monthly_Income                    0
FICO_Range                        0
Open_CREDIT_Lines                 0
Revolving_CREDIT_Balance          0
Inquiries_in_the_Last_6_Months    0
Employment_Length                 0
dtype: int64

### <center>BUSINESS PROBLEM-1</center>

**BACKGROUND:** 

The Lending Club is a peer-to-peer lending site where members make loans to each other. The site makes anonymized data on loans and borrowers publicly available.

**BUSINESS PROBLEM:**


Using lending club loans data, the team would like to test below hypothesis on how different factors effecing each other (Hint: You may leverage hypothesis testing using statistical tests)

#### (a.) Intrest rate is varied for different loan amounts (Less intrest charged for high loan amounts)

#####  1. NULL hypothesis , H0 :

                Interest rate & Loan amount are independent variables.
                Interest rate does not vary for different loan amounts.(Less interest is not charged for high loan amounts).


#####  2. ALTERNATIVE hypothesis , Ha :

                Interest rate & Loan amount are dependent variables.
                Intrest rate is varied for different loan amounts (Less intrest charged for high loan amounts).

##### 3. CI & p-values :

                CI= 95% , p-value=0.05

##### 4. Perform the test :

                pearsonr()

In [324]:
stats.pearsonr(loans.Amount_Requested , loans.Interest_Rate)

PearsonRResult(statistic=0.3324464059665682, pvalue=1.4148683739785434e-65)

##### 5. Rejection Rule :

            New P_value < 0.05 
            hence, we Reject NULL Hypothesis.

##### 6. Business conclusion :

            Interest rate is varied for different loan amounts (Less interest charged for high loan amounts).

#### (b.) Loan length is directly effecting intrest rate.

#####  1. NULL hypothesis , H0 :

                u1 = u2
                Loan length is not directly effecting intrest rate.
                There is no difference in interest rates across different loan lengths.

#####  2. ALTERNATIVE hypothesis , Ha :

                u1 <> u2
                Loan length is directly effecting intrest rate.
                There is a significant difference in interest rates for different loan lengths.

##### 3. CI & p-values :

                CI= 95% , p-value=0.05

##### 4. Perform the test :

                t-test

In [338]:
m1= loans.loc[loans.Loan_Length.str.contains('36 Months', case= False),'Interest_Rate']
m2= loans.loc[loans.Loan_Length.str.contains('60 Months', case= False),'Interest_Rate']
print('Interest Rate Mean of 36 Months Long Length : ', m1.mean())
print('Interest Rate Mean of 60 Months Long Length : ', m2.mean())

Interest Rate Mean of 36 Months Long Length :  12.127141791301895
Interest Rate Mean of 60 Months Long Length :  16.407463503649634


In [340]:
stats.ttest_ind(m1 , m2)

TtestResult(statistic=-23.41049023613487, pvalue=9.549596064332559e-110, df=2498.0)

##### 5. Rejection Rule :

            New P_value < 0.05 
            hence, we Reject NULL Hypothesis.

##### 6. Business conclusion :

            longer loan terms are associated with higher interest rates -----> Loan length is directly effecting intrest rate.

#### (c.) Interest rate varies for different purpose of loans

#####  1. NULL hypothesis , H0 :

                u1 = u2 = u3 = u4 ......
                Interest rate doest not varry for different purpose of loans.

#####  2. ALTERNATIVE hypothesis , Ha :

                u1 <> u2 <> u3 <> u4.....
                Interest rate varies for different purpose of loans.

##### 3. CI & p-values :

                CI= 95% , p-value=0.05

##### 4. Perform the test :

                f-test

In [377]:
group= []
for i , name in loans.groupby('Loan_Purpose'):
    group.append(name['Interest_Rate'].values)

In [379]:
stats.f_oneway(*group)

F_onewayResult(statistic=7.4826387840974355, pvalue=1.1599665535989115e-14)

##### 5. Rejection Rule :

            New P_value < 0.05 
            hence, we Reject NULL Hypothesis.

##### 6. Business conclusion :

             Interest rate varies for different purpose of loans.

#### (d.) There is relationship between FICO scores and Home Ownership. It means that, People with owning home will have high FICO scores.

#####  1. NULL hypothesis , H0 :

                There is no relationship between FICO scores and Home Ownership.

#####  2. ALTERNATIVE hypothesis , Ha :

                There is relationship between FICO scores and Home Ownership.

##### 3. CI & p-values :

                CI= 95% , p-value=0.05

##### 4. Perform the test :

                chi-squared test

In [406]:
observed= pd.crosstab(index= loans.FICO_Range , columns= loans.Home_Ownership)
observed

Home_Ownership  MORTGAGE  NONE  OTHER  OWN  RENT
FICO_Range                                      
640-644                0     0      0    1     4
645-649                1     0      0    0     2
650-654                1     0      0    0     0
655-659                0     0      1    0     3
660-664               41     0      0   17    67
665-669               52     0      0   18    75
670-674               68     0      1   13    91
675-679               79     0      1   11    75
680-684               55     0      2    9    91
685-689               61     0      0   10    65
690-694               55     0      0   10    75
695-699               68     0      0   10    75
700-704               60     0      0   12    59
705-709               66     0      0   12    56
710-714               55     0      0   10    47
715-719               46     0      0    2    45
720-724               49     0      0    1    64
725-729               48     0      0    5    41
730-734               46     0      0    7    41
735-739               30     0      0    6    29
740-744               26     0      0    4    23
745-749               26     0      0    7    21
750-754               32     0      0    4    25
755-759               28     0      0    6    12
760-764               28     0      0    4    14
765-769               22     0      0    3    11
770-774               12     0      0    1     4
775-779               13     0      0    3     6
780-784               21     0      0    3     4
785-789               12     0      0    2     5
790-794               15     0      0    1     4
795-799                7     0      0    1     5
800-804                7     1      0    3     2
805-809                6     0      0    2     3
810-814                5     0      0    1     2
815-819                6     0      0    0     0
820-824                0     0      0    1     0
830-834                1     0      0    0     0

In [408]:
stats.chi2_contingency(observed)

Chi2ContingencyResult(statistic=473.564802816852, pvalue=1.0065160858472195e-35, dof=148, expected_freq=array([[2.29600e+00, 2.00000e-03, 1.00000e-02, 4.00000e-01, 2.29200e+00],
       [1.37760e+00, 1.20000e-03, 6.00000e-03, 2.40000e-01, 1.37520e+00],
       [4.59200e-01, 4.00000e-04, 2.00000e-03, 8.00000e-02, 4.58400e-01],
       [1.83680e+00, 1.60000e-03, 8.00000e-03, 3.20000e-01, 1.83360e+00],
       [5.74000e+01, 5.00000e-02, 2.50000e-01, 1.00000e+01, 5.73000e+01],
       [6.65840e+01, 5.80000e-02, 2.90000e-01, 1.16000e+01, 6.64680e+01],
       [7.94416e+01, 6.92000e-02, 3.46000e-01, 1.38400e+01, 7.93032e+01],
       [7.62272e+01, 6.64000e-02, 3.32000e-01, 1.32800e+01, 7.60944e+01],
       [7.20944e+01, 6.28000e-02, 3.14000e-01, 1.25600e+01, 7.19688e+01],
       [6.24512e+01, 5.44000e-02, 2.72000e-01, 1.08800e+01, 6.23424e+01],
       [6.42880e+01, 5.60000e-02, 2.80000e-01, 1.12000e+01, 6.41760e+01],
       [7.02576e+01, 6.12000e-02, 3.06000e-01, 1.22400e+01, 7.01352e+01],
       [

##### 5. Rejection Rule :

            New P_value < 0.05 
            hence, we Reject NULL Hypothesis.

##### 6. Business conclusion :

             Therefore,we can say that there is relationship between FICO scores and Home Ownership with 95% accuracy.

### <center>BUSINESS PROBLEM - 2</center>

**BACKGROUND:**

When an order is placed by a customer of a small manufacturing company, a
price quote must be developed for that order. Because each order is unique, quotes must be
established on an order-by-order basis by a pricing expert. The price quote process is laborintensive,
as prices depend on many factors such as the part number, customer, geographic
location, market, and order volume. The sales department manager is concerned that the pricing
process is too complex, and that there might be too much variability in the quoted prices. An
improvement team is tasked with studying and improving the pricing process.
After interviewing experts to develop a better understanding of the current process, the team
designed a study to determine if there is variability between pricing experts. That is, do different
pricing experts provide different price quotes? Two randomly selected pricing experts, Mary and
Barry, were asked to independently provide prices for twelve randomly selected orders. Each
expert provided one price for each of the twelve orders.


**BUSINESS PROBLEM:** 

We would like to assess if there is any difference in the average
price quotes provided by Mary and Barry.

In [422]:
price_quotes= pd.read_csv(r"C:\Users\hp\Downloads\Python Case Study\4. Basic Statistics - Hypothesis Testing\Price_Quotes.csv")
price_quotes

Order_Number  Barry_Price  Mary_Price
0              1          126         114
1              2          110         118
2              3          138         114
3              4          142         111
4              5          146         129
5              6          136         119
6              7           94          97
7              8          103         104
8              9          140         127
9             10          152         133
10            11          108         103
11            12           97         108

#####  1. NULL hypothesis , H0 :

                u1 = u2
                there is not any difference in the average price quotes provided by Mary and Barry.

#####  2. ALTERNATIVE hypothesis , Ha :

                u1 <> u2
                there is any difference in the average price quotes provided by Mary and Barry.

##### 3. CI & p-values :

                CI= 95% , p-value=0.05

##### 4. Perform the test :

                t-test

In [429]:
print('Average Price of the Barry : ', price_quotes.Barry_Price.mean())
print('Average Price of the Mary : ', price_quotes.Mary_Price.mean())

Average Price of the Barry :  124.33333333333333
Average Price of the Mary :  114.75


In [452]:
stats.ttest_rel(price_quotes.Barry_Price , price_quotes.Mary_Price)

TtestResult(statistic=2.5213765108923494, pvalue=0.02840588045242053, df=11)

##### 5. Rejection Rule :

            New P_value < 0.05 
            hence, we Reject NULL Hypothesis.

##### 6. Business conclusion :

            There is difference between in the average price quotes provided by Mary and Barry.

### <center>BUSINESS PROBLEM-3:</center>

**BUSINESS PROBLEM:** 

Determine what effect, if any, the reengineering effort had on the
incidence behavioral problems and staff turnover. i.e To determine if the reengineering effort
changed the critical incidence rate. Is there evidence that the critical incidence rate
improved?

In [459]:
treatment = pd.read_csv(r"C:\Users\hp\Downloads\Python Case Study\4. Basic Statistics - Hypothesis Testing\Treatment_Facility.csv")
treatment.head()

Month Reengineer  Employee_Turnover       VAR4        VAR5
0      1      Prior             0.0000  24.390244   42.682927
1      2      Prior             6.0606  19.354839   25.806452
2      3      Prior            12.1212  35.087719  146.198830
3      4      Prior             3.3333  18.404908  110.429448
4      5      Prior            12.9032  17.964072   23.952096

In [461]:
treatment.isna().sum()

Month                0
Reengineer           0
Employee_Turnover    0
VAR4                 0
VAR5                 0
dtype: int64

In [465]:
treatment.dtypes

Month                  int64
Reengineer            object
Employee_Turnover    float64
VAR4                 float64
VAR5                 float64
dtype: object

In [471]:
treatment.rename(columns= {'VAR4':'TRPF(%)' , 'VAR5':'CI(%)'}, inplace= True)
treatment.head()

Month Reengineer  Employee_Turnover    TRPF(%)       CI(%)
0      1      Prior             0.0000  24.390244   42.682927
1      2      Prior             6.0606  19.354839   25.806452
2      3      Prior            12.1212  35.087719  146.198830
3      4      Prior             3.3333  18.404908  110.429448
4      5      Prior            12.9032  17.964072   23.952096

In [473]:
treatment.Reengineer.value_counts()

Reengineer
Prior    13
Post      7
Name: count, dtype: int64

#####  1. NULL hypothesis , H0 :

                u1 = u2
                The reengineering effort not changed the critical incidence rate ------> the critical incidence rate is not improved

#####  2. ALTERNATIVE hypothesis , Ha :

                u1 <> u2
                The reengineering effort changed the critical incidence rate --------> the critical incidence rate improved

##### 3. CI & p-values :

                CI= 95% , p-value=0.05

##### 4. Perform the test :

                t-test

In [478]:
m1= treatment.loc[treatment.Reengineer == 'Prior', 'CI(%)']
m2= treatment.loc[treatment.Reengineer == 'Post', 'CI(%)']
print('CI Mean of the Prior Reengineer : ', m1.mean())
print('CI Mean of the Post Reengineer : ', m2.mean())

CI Mean of the Prior Reengineer :  53.887906321846145
CI Mean of the Post Reengineer :  23.34971927988571


In [487]:
stats.ttest_ind(m1 , m2)

TtestResult(statistic=1.627914425352865, pvalue=0.12091989189884148, df=18.0)

##### 5. Rejection Rule :

            New P_value > 0.05 
            hence, we will not Reject NULL Hypothesis.

##### 6. Business conclusion :

            The critical incidence rate is not improved after the POST reengineer.

### <center>BUSINESS PROBLEM-4</center>

**BUSINESS PROBLEM:** 

We will focus on the prioritization system. If the system is working, then
high priority jobs, on average, should be completed more quickly than medium priority jobs,
and medium priority jobs should be completed more quickly than low priority jobs. Use the
data provided to determine whether this is, in fact, occurring.

In [704]:
priority_ass= pd.read_csv(r"C:\Users\hp\Downloads\Python Case Study\4. Basic Statistics - Hypothesis Testing\Priority_Assessment.csv")
priority_ass

Days Priority
0     3.3     High
1     7.9   Medium
2     0.3     High
3     0.7   Medium
4     8.6   Medium
..    ...      ...
637   2.5      Low
638   0.3     High
639   0.3   Medium
640   1.3   Medium
641   0.5     High

[642 rows x 2 columns]

#####  1. NULL hypothesis , H0 :

                u1 = u2 =u3
                All priority levels have the same average completion time --------> The system is not working.

#####  2. ALTERNATIVE hypothesis , Ha :

                u1 <> u2 <>u3
                At least one group has a different mean ------> The new system is working.

##### 3. CI & p-values :

                CI= 95% , p-value=0.05

##### 4. Perform the test :

                f-test | ANOVA test

In [707]:
m1= priority_ass.loc[priority_ass.Priority == 'High' , 'Days']
m2= priority_ass.loc[priority_ass.Priority == 'Medium' , 'Days']
m3= priority_ass.loc[priority_ass.Priority == 'Low' , 'Days']
print('Mean days of the High priorit work : ', m1.mean())
print('Mean days of the Medium priorit work : ', m2.mean())
print('Mean days of the Low priorit work : ', m3.mean())

Mean days of the High priorit work :  3.0236196319018402
Mean days of the Medium priorit work :  2.5
Mean days of the Low priorit work :  4.228358208955224


In [709]:
stats.f_oneway(m1 , m2 , m3)

F_onewayResult(statistic=1.812311010076072, pvalue=0.16411459461716182)

##### 5. Rejection Rule :

            New P_value > 0.05 
            hence, we will not Reject NULL Hypothesis.

##### 6. Business conclusion :

             The system is not working.
             There is no statistically significant evidence that job priority affects completion time.

### <center> BUSINESS PROBLEM-5 </center>

BUSINESS PROBLEM: Use the survey results to address the following questions
* What is the overall level of customer satisfaction?
* What factors are linked to satisfaction?
* What is the demographic profile of Film on the Rocks patrons?
* In what media outlet(s) should the film series be advertised?

In [713]:
films= pd.read_csv(r"C:\Users\hp\Downloads\Python Case Study\4. Basic Statistics - Hypothesis Testing\Films.csv")
films

_rowstate_                    Movie  Gender Marital_Status  Sinage  \
0             0  Ferris Buellers Day Off  Female        Married     2.0   
1             0  Ferris Buellers Day Off  Female         Single     1.0   
2             0  Ferris Buellers Day Off    Male        Married     2.0   
3             0  Ferris Buellers Day Off  Female        Married     1.0   
4             0  Ferris Buellers Day Off  Female        Married     1.0   
..          ...                      ...     ...            ...     ...   
325           0               Old School       2              2     1.0   
326           0               Old School       1              1     2.0   
327           0               Old School       2              1     2.0   
328           0               Old School       1              1     1.0   
329           0               Old School       2              2     2.0   

     Parking  Clean  Overall  Age  Income Hear_About  
0        2.0    2.0      2.0  3.0     1.0          5  
1        1.0    1.0      1.0  2.0     1.0          5  
2        4.0    3.0      2.0  4.0     1.0          5  
3        3.0    2.0      2.0  4.0     1.0          5  
4        1.0    1.0      1.0  3.0     3.0          1  
..       ...    ...      ...  ...     ...        ...  
325      2.0    1.0      1.0  2.0     1.0          1  
326      2.0    2.0      2.0  3.0     3.0          5  
327      1.0    1.0      2.0  2.0     2.0          5  
328      1.0    1.0      1.0  2.0     1.0          3  
329      2.0    3.0      1.0  2.0     1.0          5  

[330 rows x 11 columns]

In [715]:
films.groupby('Marital_Status').Gender.count()

Marital_Status
1           88
2          208
Married     12
Single      18
Slngle       2
Name: Gender, dtype: int64

In [717]:
films['Marital_Status']= films.Marital_Status.replace('Slngle','Single')

In [719]:
films.groupby('Marital_Status').Gender.count()

Marital_Status
1           88
2          208
Married     12
Single      20
Name: Gender, dtype: int64

In [721]:
films.isna().sum()

_rowstate_         0
Movie              0
Gender             0
Marital_Status     2
Sinage             2
Parking            2
Clean              3
Overall            2
Age                2
Income            16
Hear_About         7
dtype: int64

In [723]:
films.duplicated().sum()

46

In [725]:
films.drop_duplicates(inplace=True)

In [727]:
films.duplicated().sum()

0

In [729]:
films['Marital_Status']= films.Marital_Status.replace({'1':'Married','2':'Single'})

In [731]:
films['Gender'] = films.Gender.replace({'1':'Male','2':'Female'})

In [733]:
films

_rowstate_                    Movie  Gender Marital_Status  Sinage  \
0             0  Ferris Buellers Day Off  Female        Married     2.0   
1             0  Ferris Buellers Day Off  Female         Single     1.0   
2             0  Ferris Buellers Day Off    Male        Married     2.0   
3             0  Ferris Buellers Day Off  Female        Married     1.0   
4             0  Ferris Buellers Day Off  Female        Married     1.0   
..          ...                      ...     ...            ...     ...   
325           0               Old School  Female         Single     1.0   
326           0               Old School    Male        Married     2.0   
327           0               Old School  Female        Married     2.0   
328           0               Old School    Male        Married     1.0   
329           0               Old School  Female         Single     2.0   

     Parking  Clean  Overall  Age  Income Hear_About  
0        2.0    2.0      2.0  3.0     1.0          5  
1        1.0    1.0      1.0  2.0     1.0          5  
2        4.0    3.0      2.0  4.0     1.0          5  
3        3.0    2.0      2.0  4.0     1.0          5  
4        1.0    1.0      1.0  3.0     3.0          1  
..       ...    ...      ...  ...     ...        ...  
325      2.0    1.0      1.0  2.0     1.0          1  
326      2.0    2.0      2.0  3.0     3.0          5  
327      1.0    1.0      2.0  2.0     2.0          5  
328      1.0    1.0      1.0  2.0     1.0          3  
329      2.0    3.0      1.0  2.0     1.0          5  

[284 rows x 11 columns]

#### (a.) What is the overall level of customer satisfaction?

In [738]:
films.Overall.value_counts()

Overall
2.0    134
1.0    131
3.0     12
5.0      4
4.0      1
Name: count, dtype: int64

##### Overall level of consumer satisfaction is mostly 2 and 1 , that is GOOD and EXCELLENT.

#### (b.) What factors are linked to satisfaction?

#####  1. NULL hypothesis , H0 :

                There is no relationship between Overall and Clean.

#####  2. ALTERNATIVE hypothesis , Ha :

                There is relationship between Overall and Clean.

##### 3. CI & p-values :

                CI= 95% , p-value=0.05

##### 4. Perform the test :

                chi-squared test

In [753]:
observed= pd.crosstab(index= films.Overall , columns= films.Clean)
observed

Clean    1.0  2.0  3.0  4.0  5.0
Overall                         
1.0       85   39    5    1    1
2.0       32   84   14    3    0
3.0        3    5    4    0    0
4.0        0    1    0    0    0
5.0        2    1    0    0    1

In [755]:
stats.chi2_contingency(observed)

Chi2ContingencyResult(statistic=92.88713668393578, pvalue=7.327720046056968e-13, dof=16, expected_freq=array([[5.68754448e+01, 6.06049822e+01, 1.07224199e+01, 1.86476868e+00,
        9.32384342e-01],
       [5.77437722e+01, 6.15302491e+01, 1.08861210e+01, 1.89323843e+00,
        9.46619217e-01],
       [5.20996441e+00, 5.55160142e+00, 9.82206406e-01, 1.70818505e-01,
        8.54092527e-02],
       [4.34163701e-01, 4.62633452e-01, 8.18505338e-02, 1.42348754e-02,
        7.11743772e-03],
       [1.73665480e+00, 1.85053381e+00, 3.27402135e-01, 5.69395018e-02,
        2.84697509e-02]]))

##### 5. Rejection Rule :

            New P_value < 0.05 
            hence, we Reject NULL Hypothesis.

##### 6. Business conclusion :

             Therefore,we can say that There is relationship between Overall and Clean with 95% accuracy.

#####  1. NULL hypothesis , H0 :

                There is no relationship between Overall and Parking.

#####  2. ALTERNATIVE hypothesis , Ha :

                There is relationship between Overall and Parking.

##### 3. CI & p-values :

                CI= 95% , p-value=0.05

##### 4. Perform the test :

                chi-squared test

In [768]:
observed= pd.crosstab(index= films.Overall , columns= films.Parking)
observed

Parking  1.0  2.0  3.0  4.0  5.0
Overall                         
1.0       86   39    4    2    0
2.0       24   86   19    3    1
3.0        2    3    5    1    1
4.0        0    0    0    0    1
5.0        1    1    0    1    1

In [770]:
stats.chi2_contingency(observed)

Chi2ContingencyResult(statistic=181.10964214817002, pvalue=5.046695211640295e-30, dof=16, expected_freq=array([[5.26797153e+01, 6.01387900e+01, 1.30533808e+01, 3.26334520e+00,
        1.86476868e+00],
       [5.34839858e+01, 6.10569395e+01, 1.32526690e+01, 3.31316726e+00,
        1.89323843e+00],
       [4.82562278e+00, 5.50889680e+00, 1.19572954e+00, 2.98932384e-01,
        1.70818505e-01],
       [4.02135231e-01, 4.59074733e-01, 9.96441281e-02, 2.49110320e-02,
        1.42348754e-02],
       [1.60854093e+00, 1.83629893e+00, 3.98576512e-01, 9.96441281e-02,
        5.69395018e-02]]))

##### 5. Rejection Rule :

            New P_value < 0.05 
            hence, we Reject NULL Hypothesis.

##### 6. Business conclusion :

             Therefore,we can say that There is relationship between Overall and Parking with 95% accuracy.

#####  1. NULL hypothesis , H0 :

                There is no relationship between Overall and Sinage.

#####  2. ALTERNATIVE hypothesis , Ha :

                There is relationship between Overall and Sinage.

##### 3. CI & p-values :

                CI= 95% , p-value=0.05

##### 4. Perform the test :

                chi-squared test

In [777]:
observed= pd.crosstab(index= films.Overall , columns= films.Sinage)
observed

Sinage   1.0  2.0  3.0  4.0  5.0
Overall                         
1.0       72   42   12    2    2
2.0       19   87   22    4    2
3.0        0    4    8    0    0
4.0        0    0    1    0    0
5.0        1    1    1    0    1

In [779]:
stats.chi2_contingency(observed)

Chi2ContingencyResult(statistic=96.6900409305424, pvalue=1.4397985934919274e-13, dof=16, expected_freq=array([[4.25622776e+01, 6.19928826e+01, 2.03558719e+01, 2.77580071e+00,
        2.31316726e+00],
       [4.38718861e+01, 6.39003559e+01, 2.09822064e+01, 2.86120996e+00,
        2.38434164e+00],
       [3.92882562e+00, 5.72241993e+00, 1.87900356e+00, 2.56227758e-01,
        2.13523132e-01],
       [3.27402135e-01, 4.76868327e-01, 1.56583630e-01, 2.13523132e-02,
        1.77935943e-02],
       [1.30960854e+00, 1.90747331e+00, 6.26334520e-01, 8.54092527e-02,
        7.11743772e-02]]))

##### 5. Rejection Rule :

            New P_value < 0.05 
            hence, we Reject NULL Hypothesis.

##### 6. Business conclusion :

             Therefore,we can say that There is relationship between Overall and Sinage with 95% accuracy.

#### ---Chi_square shows that there is strong releanship between OVERALL and ALL OF THEM (PARKING, CLEAN, SINAGE), which affecting the overall satisfaction level.

#### (c.) What is the demographic profile of Film on the Rocks patrons?

In [854]:
gender_per= pd.DataFrame(round(films.Gender.value_counts()/films.Gender.count()*100,2)).rename(columns={'count':'%age'})
gender_per

%age
Gender       
Female  62.68
Male    37.32

In [850]:
martial_sta= pd.DataFrame(round(films.Marital_Status.value_counts()/films.Marital_Status.count()*100,2)).rename(columns={'count':'%age'})
martial_sta

%age
Marital_Status       
Single          68.09
Married         31.91

In [838]:
income_sta= pd.DataFrame(films.Income.value_counts()).rename(columns={'count':'No_of_Profiles'})
income_sta

No_of_People
Income              
1.0              118
3.0               78
2.0               74

In [846]:
age_sta= pd.DataFrame(films.Age.value_counts()).rename(columns={'count':'No_of_Profiles'})
age_sta

No_of_Profiles
Age                
2.0             145
3.0             105
1.0              22
4.0              10

##### 1. There are 62.7% of FEMALES profiles of Film on the Rocks patrons

##### 2. There are 68.09% of SINGLE profiles of Film on the Rocks patrons

##### 3. Most people are from Age group 2.0 i.e.(13-30) in the profiles of Film on the Rocks patrons

##### 4. Most people have income group of 1.0 i.e.(LESS THAN 50,000) in the profiles of Film on the Rocks patrons

#### (d.) In what media outlet(s) should the film series be advertised?

In [872]:
films.groupby('Hear_About')[['Age']].count().rename(columns= {'Age':'No_of_profiles'}).sort_values(by= 'No_of_profiles', ascending= False)

No_of_profiles
Hear_About                
5                      185
4                       38
1                       22
2                       12
3                       12
2,5                      2
1,5                      1
3,4                      1
3,5                      1
4,5                      1
5,4                      1

since , word of mouth as 5.0 is highest one but it does not come under media outlet¶

So, we'll go with the second one which is 4.0 i.e. WEBSITE

**Therefore, film series should be advertised mainly in WEBSITE media outlet.**